woocommerce/woocommerce
# Fix: Suppress deprecated warning when applying a coupon on Cart in PHP > 8.0

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





New changes on https://github.com/woocommerce/woocommerce/pull/43872 introduced a deprecated warning when attempting to run `strlen` on a null string (billing_email in this case, which is not supplied in cart)
This PR adds a new check to prevent that, and also adds the `string` `$error_message` property to class `WC_Coupon` to prevent another warning on the same class.

Closes #45956 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the extension [Debug Log Manager](https://wordpress.org/plugins/debug-log-manager/) 
2. Add a non-existent coupon on Shortcode Cart
3. Error log on `/wp-admin/tools.php?page=debug-log-manager` should not contain deprecated warnings stating `Deprecated: strlen(): Passing null to parameter` or related to creation of a dynamic property called `error_message`



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Fix: Suppress deprecated warning when applying a coupon on Cart in PHP > 8.0


</details>
